File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let result = 0;
    let x = [];
    let notX = [];

    for (const str of s) {
        if (!x.length || x[x.length -1] === str) x.push(str);
        if (x[x.length - 1] !== str) notX.push(str);
        if (x.length === notX.length) {
            result += 1;
            x = [];
            notX = [];
        }
    }
    
    if (x.length) result++;
    
    return result;
}

File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    dict = {'A': 0}
    first = ''

    for c in s:
        if first == '':
            first += c
            dict[first] = 1
            continue
            
        if c == first:
            dict[first] += 1
        else:
            dict['A'] += 1
            
        if dict[first] == dict['A']:
            print(dict)
            answer += 1
            first = ''
            dict = {'A': 0}
    return answer + 1 if dict != {'A': 0} else answer

